diff --git a/geolite2legacy.py b/geolite2legacy.py
index e6a39f1..ce7a530 100755
--- a/geolite2legacy.py
+++ b/geolite2legacy.py
@@ -316,7 +316,7 @@ class CountryRadixTree(RadixTree):
             if location is None:
                 continue
 
-            nets = [IPNetwork(row['network'])]
+            nets = [IPNetwork(row['network'].decode('utf-8'))]
             country_iso_code = location['country_iso_code'] or location['continent_code']
             yield nets, (country_iso_code,)
 
